cleanup/servicex_storage/db_manager.py

```diff
@@ -0,0 +1,159 @@
+            )
+            rows = cursor.fetchall()
+            for row in rows:
+                # print(row)
+                datasets.append(DS(row[0], row[1], row[2], row[3]))
+            cursor.close()
+
+        except psycopg2.Error as e:
+            print(f"Error geting requests: {e}")
+        return datasets
+
+    def delete_transform_results_req_id(self, req_id):
+        try:
+            cursor = self.conn.cursor()
+            cursor.execute(
+                f"delete from transform_result where request_id='{req_id}'"
+            )
+            self.conn.commit()
+            cursor.close()
+            self.logger.info(f'deleted transform_result related to {req_id}')
+
+        except psycopg2.Error as e:
+            print(f"Error deleting transform_result for req_id: {req_id}.\n{e}")
+            self.conn.rollback()
+
+    def delete_transform_results_file_id(self, file_id):
+        pass
+
+    def delete_request(self, req):
+
+        # first remove transform_result
+        self.delete_transform_results_req_id(req.req_id)
+
+        try:
+            cursor = self.conn.cursor()
+            cursor.execute(
+                f"delete from requests where request_id='{req.req_id}'"
+            )
+            self.conn.commit()
+            cursor.close()
+            self.logger.info(f'deleted request {req.req_id}')
+
+        except psycopg2.Error as e:
+            print(f"Error deleting request: {req.req_id},\n{e}")
+            self.conn.rollback()
+
+    def delete_files(self, dataset_id):
+        pass
+
+    def delete_dataset(self, ds):
+
+        # first remove transform_result
+        # self.delete_transform_results_req_id(req.req_id)
+
+        try:
+            cursor = self.conn.cursor()
+            cursor.execute(
+                f"delete from datasets where id='{ds.ds_id}'"
+            )
+            self.conn.commit()
+            cursor.close()
+            self.logger.info(f'deleted dataset {ds.ds_id}')
+
+        except psycopg2.Error as e:
+            # print(f"Error deleting dataset: {ds.ds_id}.\n{e}")
+            self.conn.rollback()
+
+    def cleanup_db(self):
+        """
+        Clean up db
+        """
+
+        # get all the requests known to ServiceX
+        requests = self.get_servicex_requests()
+        self.logger.info('request in servicex db', extra={"requests": len(requests)})
+
+        # delete too old requests.
+        for req in requests:
+            req_age = datetime.datetime.now(
+                datetime.timezone.utc) - req.submit_time.replace(tzinfo=datetime.timezone.utc)
+            if req_age.days > max_request_age:
+                self.delete_request(req)
+
+        # get all the datasets in ServiceX
+        datasets = self.get_datasets()
+        self.logger.info('datasets in servicex db', extra={"datasets": len(datasets)})
+
+        # delete datasets not used recently
```

Comment:
Is there a possible race condition here? (between lookup of datasets and deletion, a job starts and requests the dataset that is about to be cleaned?)